PyInstaller/building/splash.py

```diff
@@ -0,0 +1,291 @@
+                are not supported.
+        kwargs:
+            Splash screen options:
+                text_rect
+                    An optional 4x integer tuple that represents a text window on the
+                    splash screen where text can be displayed. The rect defines a
+                    rectangle on the bitmap by the coordinates of its upper left and
+                    lower right corners.
+                text_color
+                    An optional RGB color of the text, which can be displayed in
+                    text_rect. In hex color format, i.e. 0xRRGGBB, where RR (red),
+                    GG (green) and BB (blue) are between 0 and 0xFF. An alpha channel
+                    is not supported.
+            Target options
+                name
+                    An optional alternative filename for the *.res file. If not specified,
+                    a name is generated.
+        """
+        super(Splash, self).__init__()
+        from ..config import CONF
+
+        # Make bitmap path relative to .spec file
+        if not os.path.isabs(bitmap):
+            bitmap = os.path.join(os.path.dirname(CONF['spec']), bitmap)
+        bitmap = os.path.normpath(bitmap)
+
+        if not os.path.exists(bitmap):
+            raise ValueError("bitmap '%s' not found" % bitmap)
+
+        self.bitmap_file = bitmap
+        self.text_rect = kwargs.pop("text_rect", (0, 0, 0,))
+        self.text_color = kwargs.pop("text_color", 0xFFFFFF)
+        self.name = kwargs.pop("name", None)
+
+        if is_win:
+            # On windows the text color is in BBGGRR format
+            self.text_color = _convert_hex_to_le(self.text_color)
+
+        # Save the generated file separately so that it is not necessary to
+        # generate the data again and again
+        if self.name is None:
+            self.name = os.path.splitext(self.tocfilename)[0] + '.res'
+
+        self._check_txt_rect()
+
+        self.__postinit__()
+
+    _GUTS = (
+        # input parameters
+        ('bitmap_file', _check_guts_eq),
+        ('text_rect', _check_guts_eq),
+        ('text_color', _check_guts_eq),
+        ('name', _check_guts_eq),
+    )
+
+    def _check_guts(self, data, last_build):
+        if Target._check_guts(self, data, last_build):
+            return True
+
+        # check if the bitmap file was modified
+        if misc.mtime(self.bitmap_file) > last_build:
+            logger.info("Building %s because file %s changed",
+                        self.tocbasename, self.bitmap_file)
+            return True
+        return False
+
+    def _check_txt_rect(self):
+        if len(self.text_rect) != 4:
+            raise ValueError("wnd_text_rect must be a tuple of four positive integers!")
+        if any([not (isinstance(a, int) and a > 0) for a in self.text_rect]):
+            raise ValueError("wnd_text_rect must be a tuple of four positive integers!")
+
+    def assemble(self):
```

Comment:
I wonder what happens if one runs a .spec-file containing  `Splash()` on e.g. GNU/Linux? I assume we should check this case somewhere.

Reply:
I left this check out because the build target Splash is largely independent from the operating system (except for a few small platform dependent adjustments to the data). So the data is passed on to the bootloader, no matter which operating system is running. If the bootloader supports the feature, it will display it, if not, they will be ignored. For this I have implemented the [message in the bootloader](https://github.com/Chrisg2000/pyinstaller/blob/splash-feature/bootloader/src/pyi_splash.c#L632 ). If this is implemented for other operating systems over time, then further changes can be made to the data.

Currently on a non-Windows system the resources would be attached but not used. Maybe a message can be displayed that the data is attached, but the target platform does not yet support this?

Reply:
I have now added a[ warning message](https://github.com/Chrisg2000/pyinstaller/blob/splash-feature/PyInstaller/building/splash.py#L216) when the user is building for an unsupported platform. Nevertheless the data will still be attached.